plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/TableStatisticsReader.java

```diff
@@ -85,35 +88,36 @@
 
     public static final String APACHE_DATASKETCHES_THETA_V1_NDV_PROPERTY = "ndv";
 
-    private final TypeManager typeManager;
-    private final ConnectorSession session;
-    private final Table icebergTable;
-
-    private TableStatisticsReader(TypeManager typeManager, ConnectorSession session, Table icebergTable)
-    {
-        this.typeManager = typeManager;
-        this.session = session;
-        this.icebergTable = icebergTable;
-    }
-
     public static TableStatistics getTableStatistics(TypeManager typeManager, ConnectorSession session, IcebergTableHandle tableHandle, Table icebergTable)
     {
-        return new TableStatisticsReader(typeManager, session, icebergTable).makeTableStatistics(tableHandle);
+        return makeTableStatistics(
+                typeManager,
+                icebergTable,
+                tableHandle.getSnapshotId(),
+                tableHandle.getEnforcedPredicate(),
+                tableHandle.getUnenforcedPredicate(),
+                isExtendedStatisticsEnabled(session));
     }
 
-    private TableStatistics makeTableStatistics(IcebergTableHandle tableHandle)
+    @VisibleForTesting
+    public static TableStatistics makeTableStatistics(
+            TypeManager typeManager,
+            Table icebergTable,
+            Optional<Long> snapshot,
+            TupleDomain<IcebergColumnHandle> enforcedConstraint,
+            TupleDomain<IcebergColumnHandle> unenforcedConstraint,
+            boolean extendedStatisticsEnabled)
     {
-        if (tableHandle.getSnapshotId().isEmpty()) {
+        if (snapshot.isEmpty()) {
             // No snapshot, so no data.
             return TableStatistics.builder()
                     .setRowCount(Estimate.of(0))
                     .build();
         }
-        long snapshotId = tableHandle.getSnapshotId().get();
-
-        TupleDomain<IcebergColumnHandle> enforcedPredicate = tableHandle.getEnforcedPredicate();
+        long snapshotId = snapshot.get();
 
-        if (enforcedPredicate.isNone()) {
+        TupleDomain<IcebergColumnHandle> effectivePredicate = enforcedConstraint.intersect(unenforcedConstraint);
```

Comment:
let's capture some info from https://github.com/trinodb/trino/pull/16244#discussion_r1121142838 conversation
like that this matches split gen

Reply:
I tried to sum it up in the commit message, let me know if theres anything missing.

Reply:
> I tried to sum it up in the commit message,

code comments are easier to notice (and serve slightly different purpose)

Reply:
Added a code comment as well